packages/firebase_auth/ios/Classes/FirebaseAuthPlugin.m

```diff
@@ -116,6 +122,31 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                                         completion:^(FIRUser *user, NSError *error) {
                                           [self sendResult:result forUser:user error:error];
                                         }];
+  } else if ([@"signInWithCustomToken" isEqualToString:call.method]) {
+    NSString *token = call.arguments[@"token"];
+    [[FIRAuth auth] signInWithCustomToken:token
+                               completion:^(FIRUser *user, NSError *error) {
+                                 [self sendResult:result forUser:user error:error];
+                               }];
+
+  } else if ([@"startListeningAuthState" isEqualToString:call.method]) {
+    NSNumber *identifier = [NSNumber numberWithInteger:call.arguments[@"id"]];
+
+    FIRAuthStateDidChangeListenerHandle listener = [[FIRAuth auth]
+        addAuthStateDidChangeListener:^(FIRAuth *_Nonnull auth, FIRUser *_Nullable user) {
+          [self.channel invokeMethod:@"onAuthStateChanged" arguments:nil];
```

Comment:
I think you need to pass two arguments here, the user `NSDictionary` as well as the identifier

Reply:
Can you double check that this is right? I think you need to concert the user to an `NSDictionary` and pass it

Reply:
yes indeed, forgot that step